microcosm_pubsub/chain/chain.py

```diff
@@ -0,0 +1,70 @@
+from microcosm_pubsub.chain.context import SafeContext
+from microcosm_pubsub.chain.context_decorators import (
+    get_from_context,
+    save_to_context,
+    save_to_context_by_func_name,
+    temporarily_replace_context_keys,
+)
+
+
+class Chain:
+    """
+    Structure that contains callable functions that shares the same context.
+
+    """
+    def __init__(self, *args):
+        """
+        Build a chain that can be resolved multiple times
+        :param *args: callable functions, Chain statemnts or other Chains
+
+        """
+        self.links = list(args)
+
+    @property
+    def context_decorators(self):
+        """
+        Decorators to apply to the chain links
+
+        """
+        return [
+            # Order matter - get_from_context should be first
+            get_from_context,
+            temporarily_replace_context_keys,
+            save_to_context,
+            save_to_context_by_func_name,
+        ]
+
+    def create_context(self):
+        """
+        Context to create
+
+        """
+        return SafeContext()
```

Comment:
It's somewhat more common to define a function/property with the `class` of the context type